doc/source/security/encryption-functions.rst

```diff
@@ -0,0 +1,431 @@
+
+   * - Function Name
+   * - :ref:`asym-decrypt`
+   * - :ref:`asym-derive`
+   * - :ref:`asym-encrypt`
+   * - :ref:`asym-sign`
+   * - :ref:`asym-verify`
+   * - :ref:`create-priv-key`
+   * - :ref:`create-pub-key`
+   * - :ref:`create-dh-param`
+   * - :ref:`create-digest`
+
+The following table describes the :ref:`encryption-variables` which can be used to set the maximum value for a key length based on the type of encryption.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Variable Name
+   * - :ref:`dh-threshold`
+   * - :ref:`dsa-threshold`
+   * - :ref:`rsa-threshold`
+
+.. _install-functions:
+
+Install `component_encryption_udf`
+-------------------------------------------------
+
+Use the `Install Component Statement <https://dev.mysql.com/doc/refman/8.0/en/install-component.html>`__ to add the `component_encryption_udf` component. The functions and variables are available. The user-defined functions are auto-registered. There is no requirement to invoke ``CREATE FUNCTION ... SONAME ...``. Also, the :ref:`encryption-variables` are auto-registered when the component is installed.
+
+The following is an example of the installation command:
+
+.. code-block:: mysql
+
+    mysql> INSTALL COMPONENT 'file://component_encryption_udf';
+
+.. _asym-decrypt:
+
+asymmetric_decrypt(*algorithm, crypt_str, key_str*)
+-----------------------------------------------------------
+
+Decrypts an encrypted string using the algorithm and a key string.
+
+.. rubric:: Returns
+
+A plaintext as a string.
+
+.. rubric:: Parameters
+
+The following are the function's parameters:
+
+* algorithm - encryption algorithm used to encrypt the string. Supports `RSA`.
+
+* key_str - a string in the PEM format. The key string must be the following:
+
+  - Valid
+
+  - Public or private key string that corresponds with the private or public key string used with the :ref:`asymmetric_encrypt<asym-encrypt>` function.
+
+.. _asym-derive:
+
+asymmetric_derive(*pub_key_str, priv_key_str*)
+------------------------------------------------------------
+
+Derives a symmetric key using a public key generated on one side and a private key generated on another.
+
+.. rubric:: Returns
+
+A key as a binary sting.
+
+.. rubric:: Parameters
+
+The ``pub_key_str`` must be a public key in the PEM format and generated using the Diffie-Hellman (DH) algorithm.
+
+The ``priv_key_str`` must be a private key in the PEM format and generated using the Diffie-Hellman (DH) algorithm.
+
+.. _asym-encrypt:
+
+asymmetric_encrypt(*algorithm, str, key_str*)
+--------------------------------------------------------------
+
+Encrypts a string using the algorithm and a key string.
+
+.. rubric:: Returns
+
+A ciphertext as a binary string.
+
+.. rubric:: Parameters
+
+The parameters are the following:
+
+* algorithm - encryption algorithm used to encrypt the string. Supports `RSA`.
+
+* str - measured in bytes, the length of the string must not be greater than the key_str modulus length in bytes - 11
```

Comment:
maybe useful to mention why we are subtracting eleven. Inside brackets we can add this information.

11 (random bytes used for padding).

Reply:
Good idea. I would say `11 (additional bytes used for PKCS1 padding)`

Reply:
Done